components/browser/session-storage/src/main/java/mozilla/components/browser/session/storage/serialize/BrowserStateReader.kt

```diff
@@ -89,6 +96,12 @@ private fun JsonReader.browsingSession(
     }
 
     return if (tabs != null && tabs.isNotEmpty()) {
+        // Check if selected tab still exists after restoring/filtering and
+        // use most recently accessed tab otherwise.
```

Comment:
@pocmo ^^ this is most important to discuss, I think. It seems reasonable to me as we never want to end up without a selected tab, but it's a change to one our existing tests too. Before we allowed a null selectedTabId if the tab doesn't exist, now we will select the most recently accessed one.

Reply:
Yes, I think this makes sense here. We are not very strict about the selected ID, but in the restore use case I think we can pick an alternative tab.

Reply:
👍